Reagent Gun is now a child of /obj/item/gun/syringe

iskyp opened the pull request:
:cl:
code: chem_gun.dm has been entirely deleted
code: /obj/item/gun/chem has been changed to /obj/item/gun/syringe/chem, making it a child of already existing syringe guns
imageadd: the reagent gun's right hand sprites are now properly oriented.
/:cl:

resolves #35992

Commit:
reagent gun is now a child of /obj/item/gun/syringe

/obj/item/gun/syringe/chem

Cyberboss commented on code/modules/projectiles/guns/misc/syringe_gun.dm:
```diff
@@ -102,3 +102,44 @@
+
+/obj/item/gun/syringe/chem/Destroy()
+	. = ..()
+	STOP_PROCESSING(SSobj, src)
```
```dm
STOP_PROCESSING(SSobj, src)
return ..()
```

iskyp replied:
@Cyberboss addressed

Commit:
minor code cleanup

*added return ..() like cyberboss suggested, line 128

*removed code from line 129 to 130, which was a duplicate of an already existing parent proc

Commit:
pointless line remove

this literally does nothing

Cyberboss commented on code/modules/projectiles/guns/misc/syringe_gun.dm:
```diff
@@ -102,3 +102,41 @@
+	var/last_synth = 0
+
+/obj/item/gun/syringe/chem/attackby(obj/item/A, mob/user, params, show_msg = TRUE)
+	return
```
Pretty sure @kevinz000 made every attackby not do this

iskyp replied:
attackby here exists to stop people from loading syringes into the gun, since you're not supposed to be able to do so

ExcessiveUseOfCobblestone replied:
attackby is suppose to send signals to components so long-term this should probably be changed.

AnturK replied:
Still needs changing.

Commit:
all of the stuff inside this () is useless

like, completely pointless
*oh, and the gun starts fully loaded now like it should

Cyberboss commented on code/modules/projectiles/guns/misc/syringe_gun.dm:
```diff
@@ -113,12 +113,13 @@
 	var/last_synth = 0
 
-/obj/item/gun/syringe/chem/attackby(obj/item/A, mob/user, params, show_msg = TRUE)
+/obj/item/gun/syringe/chem/attackby()
```
no, we don't change override parameters because it propagates misinformation about the proc. Initialize and Destroy are the only exceptions

Commit:
no more overwriting parameters

ExcessiveUseOfCobblestone commented:
What's the similarities between them that warrants the subtype if you haven't changed any of the procs except for one that you have to change to split away from parent's functionality?

ExcessiveUseOfCobblestone commented:
Like what?

There's several procs that never use ..() and there's one that you explicitly set to be unlike the parent?

ExcessiveUseOfCobblestone commented:
I see now that it shares some of the variables. My bad.

Armhulen commented:
@PKPenguin321 ?

PKPenguin321 commented:
what

Jalleo commented on code/modules/projectiles/guns/misc/syringe_gun.dm:
```diff
@@ -102,3 +102,42 @@
+		chambered.newshot()
+
+/obj/item/gun/syringe/chem/process()
+	if(syringes.len >= max_syringes)
```
Change this so instead it checks if its at max now below when it adds it and remove from the objprocess then readds when next fired.

ExcessiveUseOfCobblestone replied:
He's telling you to constantly turn off processing on max (as opposed to just returning) and then turn back on processing when you fire.

Is it worth it? dunno.

Jalleo reviewed (changes requested):
Reduces overhead ticks from this if what I ask is done.

KorPhaeron commented:
@tgstation/commit-access please review or kill this its been a month

iskyp commented:
I'm just going to close this before it goes stale bc I don't really care to finish it
